src/CosmosLogStorage.ts

```diff
@@ -0,0 +1,235 @@
+     */
+    public async GetMany(appId?: string, packageIds?: string[], continuationToken?: string, pageSize = MAX_PAGE_SIZE): Promise<LogQueryResult> {
+
+        if (this.container) {
+            try {
+                let and = ""
+                const querySpec: Cosmos.SqlQuerySpec = {
+                    query: `SELECT * FROM c`,
+                    parameters: []
+                }
+                if (appId) {
+                    querySpec.query = querySpec.query.concat(' WHERE c.appId = @appId')
+                    querySpec.parameters!.push({ name: "@appId", value: appId })
+                    and = " AND"
+                }
+                if (packageIds && packageIds.length > 0) {
+                    querySpec.query = querySpec.query.concat(`${and} ARRAY_CONTAINS(@packageList, c.packageId)`)
+                    querySpec.parameters!.push({ name: '@packageList', value: packageIds })
+                    and = " AND"
+                }
+                if (this.deleteQueue && this.deleteQueue.length > 0) {
+                    querySpec.query = querySpec.query.concat(`${and} NOT ARRAY_CONTAINS(@logIdList, c.logDialogId)`)
+                    querySpec.parameters!.push({ name: '@logIdList', value: this.deleteQueue })
+                }
+
+                const feedOptions: Cosmos.FeedOptions = {
+                    maxItemCount: Math.min(pageSize, MAX_PAGE_SIZE),
+                    continuation: continuationToken
+                }
+
+                const feedResponse = await this.container.items.query(querySpec, feedOptions).fetchNext()
+
+                return {
+                    logDialogs: feedResponse.resources as CLM.LogDialog[],
+                    continuationToken: feedResponse.continuation
+                }
+            }
+            catch (err) {
+                CLDebug.Error(err)
+            }
+        }
+        throw new Error("Contained undefined")
+    }
+
+
+    /** Replace and exisiting log dialog */
+    public async Replace(appId: string, logDialog: CLM.LogDialog): Promise<void> {
+        if (this.container) {
+            try {
+                await this.container.item(this.GetDialogDocumentId(appId, logDialog.logDialogId), appId).replace(logDialog)
+            }
+            catch (err) {
+                CLDebug.Error(err)
+            }
+        }
+    }
+
+    /** Delete a log dialog in storage */
+    public async Delete(appId: string, logDialogId: string): Promise<void> {
+        if (this.container) {
+            try {
+                await this.container.item(this.GetDialogDocumentId(appId, logDialogId), appId).delete()
+            }
+            catch (err) {
+                CLDebug.Error(err)
+            }
+        }
+    }
+
+    /** Delete multiple log dialogs */
+    public async DeleteMany(appId: string, logDialogIds: string[]): Promise<void> {
+        if (this.container) {
+            try {
+                // Add items to existing delete queue
```

Comment:
Would this perform differently / better if you used a SQL-driven delete  ?

Reply:
Not supported by Cosmos.  Only one-by-one delete